cdap-client/src/main/java/co/cask/cdap/client/NamespaceClient.java

```diff
@@ -61,9 +62,18 @@ public NamespaceClient(ClientConfig config) {
    * @throws UnAuthorizedAccessTokenException if the request is not authorized successfully in the gateway server
    */
   public List<NamespaceMeta> list() throws IOException, UnAuthorizedAccessTokenException {
-    HttpResponse response = restClient.execute(HttpMethod.GET, config.resolveURL("namespaces"),
-                                               config.getAccessToken());
-    return ObjectResponse.fromJsonBody(response, new TypeToken<List<NamespaceMeta>>() { }).getResponseObject();
+    // TODO: remove the following apiVersion set/reset logic all APIs are migrated to v3
```

Comment:
It's possible to configure ClientConfig to use a particular API version I believe, but it's not very user-friendly. It's more of a separate concern though.